### PR TITLE
Bugzilla 2449 Dynlink

### PR DESCRIPTION
**Pull request by mjkoster**
RAML and JSON files

### CODE REVIEW

File: schemas/oic.linkstatus-rw-schema.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            "description": "The instance parameter of the dynamic link"
+		      }
+        }
+      }
+    },
+    "type": "object",
+    "allOf": [
+      {"$ref": "#/definitions/linkstatus-properties"}
+    ],
+    "required": ["linkstate", "laststatus", "linkinstance"],
```
Comment (mkettrick) [inline schemas/oic.linkstatus-rw-schema.json]: Per the CR, "maxwait" and "endtime" are also required.

File: schemas/oic.linkstatus-rw-schema.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      "linkstatus-properties": {
+        "properties": {
+          "linkstate": {
+            "type": "string",
+            "enum": [ "init", "active", "error", "timeout" ],
+            "readOnly": true,
+            "description": "The state of the dynlink"
+          },
+          "laststatus": {
+            "type": "number",
```
Comment (mkettrick) [inline schemas/oic.linkstatus-rw-schema.json]: This is defined as a "string" in the CR.

File: schemas/oic.linkstatus-rw-schema.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            "enum": [200, 201, 400, 403, 404, 405, 406, 408, 409, 410, 415, 418, 429, 500, 501, 502, 503, 504],
```
Comment (mkettrick) [inline schemas/oic.linkstatus-rw-schema.json]: There are more values than just these, right?  Do we really want to enumerate all possible values?  We have other strings that convey the last error code like for the OCF Cloud Resources and they are not enumerations but just a "string".

File: schemas/oic.linkstatus-rw-schema.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            "readOnly": true,
+            "description": "Last status code in http equivalent"
+		      },
+          "endtime": {
+            "type": "string",
+            "format": "date",
```
Comment (mkettrick) [inline schemas/oic.linkstatus-rw-schema.json]: "datetime"

File: schemas/oic.linkstatus-rw-schema.json [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            "readOnly": true,
+            "description": "Time at which the link times out if it remains idle"
+		      },
+          "maxwait": {
+            "type": "number",
+            "readOnly": false,
+            "description": "Maximum idle seconds until timeout error"
+		      },
+          "linkinstance": {
+            "type": "string",
```
Comment (mkettrick) [inline schemas/oic.linkstatus-rw-schema.json]: The CR says this is a "number" but I think "integer" would be better.  Either way, I don't think it should be a string.

File: dynlink.raml [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+traits:
+  - interface:
+      queryParameters:
+        if:
+          enum: [oic.if.ll", "oic.if.b", "oic.if.baseline", "oic.if.linkupdate"]
+
+
+/DynlinkResURI?if=oic.if.baseline:
+    description: |
+      Baseline representation of a dynlink contains only its common properties
```
Comment (mkettrick) [inline dynlink.raml:29]: No.  Per the Core Spec, baseline returns ALL Properties.  Either this needs to be updated here or a CR needs to be written to redefine what baseline Interface is.

File: dynlink.raml [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+        description: |
+          Common properties of the dynlink resource type
+        responses:
+          200:
+            body:
+              application/json:
+                schema: baseline
+                example: |
+                {
+                  "rt": ["oic.r.linkupdate"],
```
Comment (mkettrick) [inline dynlink.raml]: I've never seen this Resource Type defined.  oic.if.linkupdate is an Interface name but not an "rt", right?

File: dynlink.raml [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+                schema: linklist
+                  example: |
+                    [
+                      {
+                        "href": "ocf://5c68f373-0c4c-4604-9922-db639e20533d/switch",
+                        "rel": ["boundto"],
+                        "bind": "obs",
+                        "anchor": "modeswitch",
+                        "if": ["oic.if.s"],
+                        "ins": "377fac40"
```
Comment (mkettrick) [inline dynlink.raml:68]: "ins" is defined as an integer in Collection schema, not a string.

File: dynlink.raml [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+                        "rep": {
+                          "linkstate": "active",
+                          "laststatus": "200",
+                          "endtime": "2018-06-20T07:07:03+00:00",
+                          "maxwait": 6000,
+                          "linkinstance": "ca5f1004"
+                        }
+                      }
+                    ]
+
```
Comment (mkettrick) [inline dynlink.raml:153]: Why doesn't the "batch" response include a representation of the Dynamic Link Resources?  They are in the Collection so they should be included, right?

**Comment by rabardini** [comment]
Moved to Backlog for now so closing the Pull Request.

**Pull request closed without merging**